src/main/java/io/neonbee/config/ServerConfig.java

```diff
@@ -0,0 +1,798 @@
+
+    /**
+     * The default compression level is 1, which means minimal compression, but less resource consumption.
+     */
+    public static final int DEFAULT_COMPRESSION_LEVEL = 1;
+
+    /**
+     * By default NeonBees server verticle is using decompression support.
+     */
+    public static final boolean DEFAULT_DECOMPRESSION_SUPPORTED = true;
+
+    /**
+     * The default timeout for web requests to the {@link ServerVerticle} is 30 seconds.
+     */
+    public static final int DEFAULT_TIMEOUT = 30;
+
+    /**
+     * The default name to store the session information in a cookie.
+     */
+    public static final String DEFAULT_SESSION_COOKIE_NAME = "neonbee-web.session";
+
+    private static final List<EndpointConfig> DEFAULT_ENDPOINT_CONFIGS = Collections.unmodifiableList(Arrays
+            .stream(new Class<?>[] { RawEndpoint.class, ODataV4Endpoint.class, MetricsEndpoint.class })
+            .map(endpointClass -> new EndpointConfig().setType(endpointClass.getName())).collect(Collectors.toList()));
+
+    private static final ImmutableBiMap<String, String> REPHRASE_MAP = ImmutableBiMap.<String, String>builder()
+            .put("endpointConfigs", "endpoints").put("authChainConfig", "authenticationChain").build();
+
+    private int timeout = DEFAULT_TIMEOUT;
+
+    private int timeoutStatusCode = GATEWAY_TIMEOUT.code();
+
+    private SessionHandling sessionHandling = SessionHandling.NONE;
+
+    private String sessionCookieName = DEFAULT_SESSION_COOKIE_NAME;
+
+    private CorrelationStrategy correlationStrategy = CorrelationStrategy.REQUEST_HEADER;
+
+    private List<EndpointConfig> endpointConfigs = new ArrayList<>(DEFAULT_ENDPOINT_CONFIGS);
+
+    private List<AuthHandlerConfig> authChainConfig;
+
+    /**
+     * Create a default server configuration.
+     */
+    public ServerConfig() {
+        super();
+        overrideDefaults(ImmutableJsonObject.EMPTY);
+    }
+
+    /**
+     * Create a server configuration based on a given JSON object.
+     *
+     * @param json the JSON object to parse
+     */
+    public ServerConfig(JsonObject json) {
+        super(json);
+
+        JsonObject newJson = rephraseConfigNames(json.copy(), REPHRASE_MAP, true);
+        overrideDefaults(newJson);
+        ServerConfigConverter.fromJson(newJson, this);
+    }
+
+    private void overrideDefaults(JsonObject json) {
+        // we override some settings like the default port, use alpn and compression settings
+        setPort(json.getInteger(PROPERTY_PORT, DEFAULT_PORT));
+        setUseAlpn(json.getBoolean("useAlpn", DEFAULT_USE_ALPN));
+        setCompressionSupported(json.getBoolean("compressionSupported", DEFAULT_COMPRESSION_SUPPORTED));
+        setCompressionLevel(json.getInteger("compressionLevel", DEFAULT_COMPRESSION_LEVEL));
+        setDecompressionSupported(json.getBoolean("decompressionSupported", DEFAULT_DECOMPRESSION_SUPPORTED));
+    }
+
+    @Override
+    public JsonObject toJson() {
```

Comment:
I think toJson should also respect the attributes stored in super class, shouldn't it?